src/OpenIddict.Server/Internal/OpenIddictServerProvider.Session.cs

```diff
@@ -137,14 +139,12 @@ async Task<bool> ValidatePostLogoutRedirectUriAsync(string address)
                         }
                     }
 
+                    _logger.LogError("The logout request was rejected because the specified application has no logout permission: {address}.", address);
```

Comment:
At this point, it's not really "exact" to say "specified application" as end session requests don't include a `client_id` (the specification clearly sucks in this regard).

Shouldn't we keep the log message at the original place (l. 145/147) but reword it a bit to explain why it's unknown? (i.e no application with the specified `post_logout_redirect_uri` AND with a logout endpoint permission endpoint was found)